PCGen-base/code/src/java/pcgen/base/format/ArrayFormatManager.java

```diff
@@ -58,105 +72,123 @@
 	 * FormatManager and separator.
 	 * 
 	 * @param underlying
-	 *            The FormatManager representing objects contained within the
-	 *            array
-	 * @param separator
-	 *            The separator character used to parse instructions and
-	 *            separate items that will be part of the array
+	 *            The FormatManager representing objects contained within the array
+	 * @param groupSep
+	 *            The group separator character used to parse instructions and separate
+	 *            groups of lists that will be part of the array
+	 * @param listSep
+	 *            The separator character used to parse lists in the instructions and
+	 *            separate list items that will be part of the array
 	 */
-	public ArrayFormatManager(FormatManager<T> underlying, char separator)
+	public ArrayFormatManager(FormatManager<T> underlying, char groupSep, char listSep)
 	{
+		componentManager = underlying;
 		@SuppressWarnings("unchecked")
-		Class<T[]> fClass = (Class<T[]>) Array
+		Class<T[]> arrayClass = (Class<T[]>) Array
 			.newInstance(underlying.getManagedClass(), 0).getClass();
-		this.separator = separator;
-		formatClass = fClass;
-		componentManager = underlying;
+		this.formatClass = arrayClass;
+		this.groupSep = groupSep;
+		this.listSep = listSep;
 	}
 
 	/**
 	 * Converts the instructions into an array of objects. The objects referred
 	 * to in the instructions should be separated by the separator provided at
 	 * construction of this ArrayFormatManager.
 	 */
+	@SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
```

Comment:
Can the Arrays class help here?

Reply:
In this case, not that I know of.  This can't use setAll because that's not what it's doing.  It's taking an array (groups) and expanding them into a single array.  Effectively I could convert the Array to a Stream, pass it through a .flatMap and then process that, but unless someone comes from the functional world, I think that's even less readable than it is now.